python/chainer_compiler.py

```diff
@@ -162,16 +167,42 @@ def backward(self, indexes, flat_gys):
         return gxs
 
 
+def _run_translator(translator, mc, inputs):
+    if translator == 'ch2o':
+        xmodel = ch2o.compile_model(mc, inputs)
+        f = tempfile.NamedTemporaryFile(delete=False)
+        f.write(xmodel.SerializeToString())
+        f.close()
+        del xmodel
+    elif translator == 'onnx_chainer':
+        import onnx_chainer
+        f = tempfile.NamedTemporaryFile(delete=False)
+        onnx_chainer.export(mc, inputs, filename=f)
+        f.close()
+    else:
+        raise NotImplementedError('Unsupported translator:',
+                                  translator)
+
+    graph = chainer_compiler_core.load(f.name)
+    os.unlink(f.name)
+
+    return graph
+
+
 class CompiledModel(chainer.Chain):
 
     def __init__(self, model, inputs, translator='ch2o', dump_onnx=False,
-                 computation_order=None):
+                 computation_order=None,
+                 export_allocator=None,
+                 runtime_allocator=None):
         super(CompiledModel, self).__init__()
         with self.init_scope():
             self.mc = model
         self.translator = translator
         self.dump_onnx = dump_onnx
         self.computation_order = computation_order
+        self.export_allocator = export_allocator
+        self.runtime_allocator = runtime_allocator
```

Comment:
Could you add something like
```
if export_allocator is not None and runtime_allocator is None:
  raise ValueError('`runtime_allocator` must be set when `export_allocator` is set')
```
I think this combination would indicate the caller does not understand these parameters.

Reply:
Added.